src/dotnet/APIView/APIViewWeb/Managers/APIRevisionsManager.cs

```diff
@@ -512,6 +547,30 @@ public async Task SoftDeleteAPIRevisionAsync(ClaimsPrincipal user, string review
             await SoftDeleteAPIRevisionAsync(user, apiRevision);
         }
 
+        /// <summary>
+        /// Delete APIRevisions
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="reviewId"></param>
+        /// <param name="apiRevisionId"></param>
+        /// <returns></returns>
+        public async Task RestoreAPIRevisionAsync(ClaimsPrincipal user, string reviewId, string apiRevisionId)
+        {
+            var apiRevision = await _apiRevisionsRepository.GetAPIRevisionAsync(apiRevisionId: apiRevisionId);
+            ManagerHelpers.AssertAPIRevisionDeletion(apiRevision);
+            await ManagerHelpers.AssertAPIRevisionOwner(user, apiRevision, _authorizationService);
```

Comment:
Do we need two assertions here or just check revision ownership?

Reply:
Well AssertAPIRevisionDeletion check that it's a manual APIRevisions. PR revisions can be created by users other than azure-sdk